### DISCUSSION TITLE
Current problem with pre-build scripts?

### DISCUSSION THREAD

**Discussion by schox** [codemagic-ci-cd/codemagic-docs#2695] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

Pre-build script appears to have stopped working?

It was working fine yesterday and then a subsequent build generated the error:

`Error (Xcode): lib/env/envied.dart:3:6: Error: Error when reading 'lib/env/envied.g.dart': No such file or directory`

On inspection of the build script logs I see:

```
[INFO] Generating build script...
[INFO] Generating build script completed, took 176ms
[INFO] Precompiling build script......
[INFO] Precompiling build script... completed, took 2.1s
[INFO] Initializing inputs
[INFO] Building new asset graph...
[INFO] Building new asset graph completed, took 1.0s
[INFO] Checking for unexpected pre-existing outputs....
[INFO] Checking for unexpected pre-existing outputs. completed, took 0ms
[INFO] Running build...
[INFO] Running build completed, took 2ms
[INFO] Caching finalized dependency graph...
[INFO] Caching finalized dependency graph completed, took 52ms
[INFO] Succeeded after 59ms with 0 outputs (0 actions)
```

So no actual file generation appears to have happened.

Yesterday, I was using this version of envied.dart (which should get picked up by build_runner to generate envied.g.dart.

```
import 'package:envied/envied.dart';

part 'envied.g.dart';

@Envied(
  path: bool.fromEnvironment('CI') ? null : '.env.prod',
)
abstract class Env {
    @EnviedField(varName: 'ONESIGNAL_APP_ID')
    static const String ONESIGNAL_APP_ID = _Env.ONESIGNAL_APP_ID;

…

}
```

And in the Pre-build script I was just calling:

`dart run build_runner build --delete-conflicting-outputs`

Once the error occured I changed envied.dart to:

```
import 'package:envied/envied.dart';

part 'envied.g.dart';

@Envied(path: '.env.prod')
abstract class Env {
    @EnviedField(varName: 'ONESIGNAL_APP_ID')
    static const String ONESIGNAL_APP_ID = _Env.ONESIGNAL_APP_ID;

…

}
```

And tried this instead in the script:

```
#!/usr/bin/env bash

# Clean previous build artifacts
flutter clean

# Create the .env.prod file with environment variables
tee -a .env.prod << EOF
ONESIGNAL_APP_ID=$ONESIGNAL_APP_ID
RC_GOOGLE_API_KEY=$RC_GOOGLE_API_KEY
RC_APPLE_API_KEY=$RC_APPLE_API_KEY
SUPABASE_URL=$SUPABASE_URL
SUPABASE_ANON_KEY=$SUPABASE_ANON_KEY
SUPABASE_BASE_URL=$SUPABASE_BASE_URL
SENTRY_DSN=$SENTRY_DSN
EOF

# Output the .env.prod file for verification
echo "Contents of .env.prod:"
cat .env.prod

# Install dependencies
flutter pub get

# Run build_runner to generate the necessary files
dart run build_runner build --delete-conflicting-outputs
```

In both cases the project builds and funs fine locally on a simulator.

### Expected results

The build script should run build_runner and generate the necessary .g.dart file.

### Actual results

No output is generated, which gives rise to the file not found error in the build process.

**Comment by himesh-cm** [comment]
Hey @schox,
Can you please share the build ID with me? You can find it under your build details page's 'ID' section.  ﻿Navigate to the 'Builds' section on the left-hand menu, select the build you want to view, and click the copy icon.

**Reply by schox** [reply]
Build ID is 66cec7a657d16e2fb1b8631f. This is a subsequent build, but same errror. I just added --verbose to the build_runner

**Reply by himesh-cm** [reply]
Hey @schox, 
There is a known issue thread here - https://github.com/petercinibulk/envied/issues/6#issuecomment-1243434607

Please run
`flutter pub run build_runner clean`
and then
`flutter pub run build_runner build --delete-conflicting-outputs`

**Reply by schox** [reply]
I tried this before submitting this issue and it didn't work. I've tried it again and it still doesn't work. I'll keep experimenting.

**Reply by himesh-cm** [reply]
Please share the build ID with me. The one you shared earlier uses `dart` instead of `flutter`.

**Reply by schox** [reply]
Here it is: 66d1bfd2bc682c6ea1d4b060

**Reply by himesh-cm** [reply]
Hey @schox,

I noticed that your latest build was successful. Are you still in need of assistance? Also, could you please share the solution that worked for you? It might be helpful for others who encounter similar issues.

**Reply by schox** [reply, marked as answer]
Yes it is now successfully building, using both versions of the pre-build script I originally described in the issue:

I'm not certain what fixed it, but I have noticed that I occasionally need to run:

flutter pub run build_runner clean

before:

flutter pub run build_runner build --delete-conflicting-outputs

on local builds if the envied.g.dart does not get generated correctly. Perhaps this applies to Codemagic builds too. I haven't found a repeatable pattern here.

Also the only other file I edited during this period was pubspec.yaml. Maybe there was an extraneous character under here somewhere:

dev_dependencies:
  build_runner: ^2.4.9
  custom_lint: ^0.6.4
  flutter_launcher_icons: ^0.13.1
  envied_generator: ^0.5.4+1
  
Sorry to not have more a specific, repeatable solution.

**Reply by himesh-cm** [reply]
Thanks for sharing the feedback.
While I’m closing our discussion for now, if you have any other questions or suggestions, feel free to reply back or start a new discussion anytime. Have a nice day!
